lib/gprdmat.gi

```diff
@@ -406,14 +406,91 @@ end );
 InstallOtherMethod( Projection,"matrix wreath product", true,
   [ IsMatrixGroup and HasWreathProductInfo ],0,
 function( W )
-local  info,proj,H;
+local  info, degI, dimA, zero, projFunc;
   info := WreathProductInfo( W );
   if IsBound( info.projection ) then return info.projection; fi;
 
-  proj:=Error("TODO");
+  degI := info.degI;
+  dimA := info.dimA;
+  zero := Zero(info.field);
+
+  projFunc := function(x)
+    local topImages, k, l, a;
+    topImages := [];
+    for k in [1 .. degI] do
+      for l in [1 .. degI] do
+        for a in [1 .. dimA] do
+          if x[dimA * (k - 1) + a, dimA * (l - 1) + a] <> zero then
+            Add(topImages, l);
+            break;
+          fi;
+        od;
+        if Length(topImages) = k then
+          break;
+        fi;
+      od;
+      if Length(topImages) <> k then
+        return fail;
+      fi;
+    od;
+    return PermList(topImages);
+  end;
+
+  info.projection := GroupHomomorphismByFunction(W, info.groups[2], projFunc);
+  return info.projection;
+end);
+
+#############################################################################
+##
+#M  ListWreathProductElementNC( <G>, <x> )
+##
+InstallMethod( ListWreathProductElementNC, "matrix wreath product", true,
+ [ IsMatrixGroup and HasWreathProductInfo, IsObject, IsBool], 0,
+function(G, x, testDecomposition)
+  local info, degI, dimA, h, list, i, j, k, zeroMat;
+
+  info := WreathProductInfo(G);
+  degI := info.degI;
+  dimA := info.dimA;
+
+  # The top group element
+  h := x ^ Projection(G);
+  if h = fail then
+    return fail;
+  fi;
+  list := EmptyPlist(degI + 1);
+  list[degI + 1] := h;
+  if testDecomposition then
+    zeroMat := NullMat(dimA, dimA, info.field);
```

Comment:
If your goal is to let `x` be any MatrixObj, then you should use something like `ZeroMatrix(dimA, dimA, x)` here (perhaps it needs a special case for `IsPlistRep`, see e.g. [here](https://github.com/gap-packages/forms/pull/4/files#diff-ad4960ae227bb883de1bf424204b3fb9d2a4b63a9d795a278b85600e211e87d4R21-R27).